opacus/tests/privacy_engine_test.py

```diff
@@ -102,25 +102,25 @@ def _init_private_training(
         noise_multiplier: float = 1.0,
         max_grad_norm: float = 1.0,
         poisson_sampling: bool = True,
-        try_fix_incompatible_modules: bool = False,
     ):
+        privacy_engine = PrivacyEngine(secure_mode=secure_mode)
+
         model = SampleConvNet()
+        model = privacy_engine.get_compatible_module(model)
```

Comment:
I'd rather have a test both both occasions: where we call `get_compatible_module` and where we don't.
Both should work (obvs provided the model is compatible in the first place). 

The original bug was spotted by dumb luck: `dcgan` example just turned out to be the only place where we used fixing behaviour, but it wasn't designed for it.
Let's make sure we have dedicated test for this